state/execution.go

```diff
@@ -131,50 +134,53 @@ func (blockExec *BlockExecutor) CreateProposalBlock(
 		// purpose for now.
 		panic(err)
 	}
-	rawNewData := preparedProposal.GetBlockData()
 	var blockDataSize int
-	for _, tx := range rawNewData.GetTxs() {
+	for _, tx := range rpp.GetTxs() {
 		blockDataSize += len(tx)
-
-		if maxDataBytes < int64(blockDataSize) {
+		if maxDataBytes < int64(blockDataSize)-tmhash.Size {
 			panic("block data exceeds max amount of allowed bytes")
 		}
 	}
 
-	newData, err := types.DataFromProto(rawNewData)
-	if err != nil {
-		// todo(evan): see if we can get rid of this panic
-		panic(err)
+	// Celestia passes the data root back as the first transaction returned in
+	// PrepareProposal. Here it is extracted out and set in the header
```

Comment:
I'm in favor of this PR! If we adopt it, we may also want to update the line

> Note on the implementation: celestia-app computes the hash in prepare_proposal and returns it to CometBFT via [`blockData.Hash`](https://github.com/celestiaorg/celestia-app/blob/5bbdac2d3f46662a34b2111602b8f964d6e6fba5/app/prepare_proposal.go#L78) so a modification to celestia-core isn't strictly necessary but [comments](https://github.com/celestiaorg/celestia-core/blob/2ec23f804691afc196d0104616e6c880d4c1ca41/types/block.go#L1041-L1042) were added.

in the README here: https://github.com/celestiaorg/celestia-core/blob/9bfa856909861081d2bbc715ee00ed56e9fc43bb/README.md?plain=1#L13